### PR TITLE
CAMEL-16486: camel-yaml-dsl-deserializers ErrorHandlerBuilder can parse only Ref

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@squakez mind having a look ?

**Review by davsclaus — APPROVED** [review]

File: dsl/camel-yaml-dsl/camel-yaml-dsl-deserializers/src/main/java/org/apache/camel/dsl/yaml/deserializers/ErrorHandlerBuilderDeserializer.java [hunk 1]
```diff
@@ -16,44 +16,101 @@
...
-    protected boolean setProperty(ErrorHandlerBuilderRef target, String propertyKey, String propertyName, Node value) {
-        if ("ref".equals(propertyKey)) {
-            target.setRef(asText(value));
-            return true;
+            setDeserializationContext(val, dc);
+
+            switch (key) {
+                case "ref": {
+                    final ErrorHandlerBuilder builder = new ErrorHandlerBuilderRef(asText(val));
+                    return new CamelContextCustomizer() {
```
Comment (squakez) [inline dsl/camel-yaml-dsl/camel-yaml-dsl-deserializers/src/main/java/org/apache/camel/dsl/yaml/deserializers/ErrorHandlerBuilderDeserializer.java:75]: Maybe a static factory method here as you're reusing the same piece of code many times

**Review by squakez — APPROVED** [review]
LGTM, thanks!

**Review by oscerd — APPROVED** [review]

File: dsl/camel-yaml-dsl/camel-yaml-dsl-deserializers/src/main/java/org/apache/camel/dsl/yaml/deserializers/ErrorHandlerBuilderDeserializer.java (hunk 1 above)
Comment (lburgazzoli) [reply dsl/camel-yaml-dsl/camel-yaml-dsl-deserializers/src/main/java/org/apache/camel/dsl/yaml/deserializers/ErrorHandlerBuilderDeserializer.java]: done

**Pull request merged**